src/reducetensor.cpp

```diff
@@ -267,6 +285,124 @@ inline int GetReduceTensorOpId(miopenReduceTensorOp_t t)
     };
 };
 
+static std::string get_network_config_string_from_type_enums(miopenDataType_t TSrc,
+                                                             miopenDataType_t TComp,
+                                                             miopenDataType_t TDst)
+{
+    std::ostringstream outs;
+
+    outs << TSrc << TComp << TDst;
+
+    return (outs.str());
+};
+
+static std::string get_definition_string_from_type_enums(miopenDataType_t TSrc,
+                                                         miopenDataType_t TComp,
+                                                         miopenDataType_t TDst)
+{
+    std::ostringstream outs;
+
+    outs << " -DCK_PARAM_SRC_DATATYPE=" << TSrc;
+    outs << " -DCK_PARAM_DST_DATATYPE=" << TDst;
+    outs << " -DCK_PARAM_REDUCE_COMPTYPE=" << TComp;
+
+    return (outs.str());
+};
+
+static std::string get_network_config_string_from_tunable(const tunable_generic_reduction* pt)
+{
+    std::ostringstream outs;
+
+    outs << "TUN_" << pt->BlockSize << "_";
+    outs << pt->GredThreadBufferLength << "_";
+    outs << pt->GredAccessesPerThreadInBlock << "_";
+    outs << pt->GredAccessesPerThreadInWarp;
+
+    return (outs.str());
+};
+
+static std::string get_definition_string_from_tunable(const tunable_generic_reduction* pt)
+{
+    std::ostringstream outs;
+
+    outs << " -DCK_PARAM_BLOCKSIZE=" << pt->BlockSize;
+    outs << " -DCK_PARAM_THREAD_BUFFER_LENGTH=" << pt->GredThreadBufferLength;
+    outs << " -DCK_PARAM_ACCESSES_PER_THREAD_INBLOCK=" << pt->GredAccessesPerThreadInBlock;
+    outs << " -DCK_PARAM_ACCESSES_PER_THREAD_INWARP=" << pt->GredAccessesPerThreadInWarp;
+
+    return (outs.str());
+};
+
+static std::string getReductionMethodStr(ReductionMethod_t reduceImpl)
+{
+    switch(reduceImpl)
+    {
+    case Reduce_DirectThreadWise: return (std::string("threadwise"));
+    case Reduce_DirectWarpWise: return (std::string("warpwise"));
+    case Reduce_BlockWise: return (std::string("blockwise"));
+    case Reduce_MultiBlock: return (std::string("multiblock"));
+    default: MIOPEN_THROW("Invalid reduction method ID!"); break;
```

Comment:
I found that  if `throw std::runtime_error` is used instead of `MIOPEN_THROW`,  cppcheck will report error due to the `break`, but `MIOPEN_THROW` seems able to avoid that